src/VisualStudio/Core/Def/Implementation/DesignerAttribute/DesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -142,7 +143,13 @@ private async Task<bool> IsCpsProjectAsync(Project project, CancellationToken ca
 
             // CPS projects do not support designer attributes.  So we just skip these projects entirely.
             var vsWorkspace = project.Solution.Workspace as VisualStudioWorkspaceImpl;
-            var cps = await Task.Factory.StartNew(() => vsWorkspace?.IsCPSProject(project) == true, cancellationToken, TaskCreationOptions.None, this.ForegroundTaskScheduler).ConfigureAwait(false);
+
+            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, cancellationToken);
+            var cps = vsWorkspace?.IsCPSProject(project) == true;
+
+            // The remainder of this method does not need to execute on the UI thread, but it's pointless to force a
+            // context switch if the caller of this method is the UI thread.
```

Comment:
i don't understand this comment.

Reply:
➡️ The comment is taking the place of the following code, which would be an accurate preservation of the original implementation:

```csharp
await TaskScheduler.Default;
```

However, switching back to the thread pool at this location is simply an unnecessary context switch for no gain, so it was omitted during the conversion.